File: be/Controllers/StatisticsController.cs
using FA23_Convocation2023_API.Services;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace FA23_Convocation2023_API.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class StatisticsController : ControllerBase
    {
        private readonly StatisticsService _service;
        public StatisticsController(StatisticsService service)
        {
            _service = service;
        }
        [HttpGet("active-halls-summary")]
        public async Task<IActionResult> GetActiveHallSummary()
        {
            var result = await _service.GetActiveHallSummaryAsync();
            if (result == null || !result.Any())
                return NotFound(new { message = "No active hall check-ins found." });

            return Ok(new
            {
                status = StatusCodes.Status200OK,
                message = "Get active hall successfully!",
                data = result
            });
        }

        [HttpGet("hall-overview")]
        public async Task<IActionResult> GetHallOverview()
        {
            var result = await _service.GetHallOverviewAsync();
            if (result == null || !result.Any())
                return NotFound(new { message = "No hall data found." });

            return Ok(new
            {
                status = StatusCodes.Status200OK,
                message = "Get hall sumary successfully!",
                data = result
            });
        }
    }
}

File: be-nest/src/statistics/statistics.controller.ts
import { Controller, Get, NotFoundException } from '@nestjs/common';
import { ApiTags } from '@nestjs/swagger';
import { Public } from '../common/guards/public.decorator.js';
import { StatisticsService } from './statistics.service.js';

@ApiTags('statistics')
@Controller('Statistics')
@Public()
export class StatisticsController {
  constructor(private readonly statistics: StatisticsService) {}

  @Get('active-halls-summary')
  async activeHallSummary() {
    const data = await this.statistics.activeHallSummary();
    if (!data.length) throw new NotFoundException('Không tìm thấy hội trường đang mở check-in.');
    return { status: 200, message: 'Lấy hội trường đang hoạt động thành công!', data };
  }

  @Get('hall-overview')
  async hallOverview() {
    const data = await this.statistics.hallOverview();
    if (!data.length) throw new NotFoundException('Không tìm thấy dữ liệu hội trường.');
    return { status: 200, message: 'Lấy tổng quan hội trường thành công!', data };
  }
}
